### ISSUE TITLE
Percent Change

### ISSUE THREAD

**Issue by alitekdemir** [freqtrade/freqtrade-strategies#205]
Hello, how to write a strategy buying code that can keep up with the percentage change?
From what I've read from the docs I should use a dictionary for the percentage change but i can't write code. I would be very grateful if you could support me on how to write a code compatible with the past candle count and hyperopt.

**Comment by xmatthias** [comment]
Hi,

Keep in mind that this is an open source project run by volunteers in their free time. You're asking people to develop a strategy for you  - based on very vague definitions, witout any (visible) effort from your side.


There's a lot of sample strategies in this repository that you can take inspiration from.

We also try to assist people when possible, however the base assumption and demand from our side is that there is some serious effort put in upfront from the users side (research basic things yourself, show the code you have so far, ... ).


Basically, you're putting in 1 sentence - with the request "now write my strategy for me".

This type of requests will not be supported - therefore - i'll close this issue.
Feel free to open a new issue if you have very specific questions (best providing the code you have so far, and where you are stuck in particular).

**Comment by alitekdemir** [comment]
You wrote a lot of sentences. and the documentation is very rich. As I said, I just wanted you to show me the way. I respect your free software effort. I expected your guidance. Many strategies work with indicators and are very successful, but I have not seen any examples of price capturing. I would appreciate it if you could share the link to the document
I think your nerves are pretty frayed. Please note that my native language is not English. I get help from translate. You are also experts and guides. We are the user. I sincerely thank you for your prompt responses and interest.

**Comment by xmatthias** [comment]
> I think your nerves are pretty frayed

Honestly, issues like this show disrespect towards the work we put in - as you clearly didn't do any effort yourself.
Trying to get offending now will only enforce this feeling, language has however nothing to do with it (that'll be wording that might be off - not the effort that's put in).

> I expected your guidance.

You will for sure get guidance assuming some initial effort has been shown, but "i can't write code" basically means "write this for me" - which simply put - is not going to happen.

As such, letme repeat, please do your research yourself.

---

You can get the current price from [dataprovider](https://www.freqtrade.io/en/stable/strategy-customization/#orderbookpair-maximum).

how you use / interpret this is up to you. 
As you however point out, you're not a programmer, i'd suggest to get started with some other form of strategy that's usually easier to implement - as you cannot backtest strategies that use current prices.

Also, you should note that also indicators are usually based on price movements / price changes - so i'm not sure that will exclude them for what you want.

**Comment by alitekdemir** [comment]
I tried to use the dataframe by vectorizing it, but I am constantly getting stuck somewhere and getting an error message.
What I'm trying to do is compare the closing price of each bar with the previous calculated data.
I guess I won't be able to figure out vectorization.

```python
    x1 = x2= high = low = 0
    value = 5

    def calculate(price, value):
        global x1, x2, high, low
        percent = price / 100 * value
        x1 = price
        x2 = x1 + percent
        high = x1 + percent * 2
        low = x1 - percent
        return x1, x2, high, low

    def populate_indicators(self, df: df, metadata: dict) -> df:
        if ("high" not in df.columns):
            x1, x2, high, low = self.calculate(int(df["close"][:1]), (self.value))
            df['high'] = high
            df['low'] = low
            df['trend'] = 0
        elif (df["close"] <= df["low"].shift(1) ):
            x1, x2, high, low = self.calculate(int(df["close"][:1]), (self.value))
            df['high'] = high
            df['low'] = low
            df['trend'] = -1
        elif (df["close"] >= df["high"].shift(1)):
            x1, x2, high, low = self.calculate(int(df["close"][:1]), (self.value))
            df['high'] = high
            df['low'] = low
            df['trend'] = 1
        else:
            df['high'] = df["high"].shift(1)
            df['low'] = df["low"].shift(1)
            df['trend'] = 0
        return df

    def populate_high_trend(self, df: df, metadata: dict) -> df:
        df.loc[ (df['trend'] == 1), 'high'] = 1
        return df

    def populate_low_trend(self, df: df, metadata: dict) -> df:
        df.loc[ (df['trend'] == -1), 'low'] = 1
        return df
```


In fact, a version where I can keep this information in "custom_info = { } " would be much more practical, but every time I tried to check and process the last data, I encountered a problem again.
(This is old code)
```python
    settings = {"percent": 2}  # "value": 2,
    custom_info = {}

    def calculate(price, value):
        percent = price / 100
        x1 = price
        x2 = x1 + percent
        high = x1 + percent * 2
        low = x1 - percent
        return x1, x2, high, low

    def populate_indicators(self, dataframe: DataFrame, pair: str,
                            metadata: dict) -> DataFrame:
        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        last_candle = float(dataframe.iloc[-1].squeeze())
        x1, x2, high, low = self.calculate(price=last_candle,
                                           value=self.settings["percent"])
        if not metadata["pair"] in self.cx2om_info:
            self.cx2om_info[metadata["pair"]] = {
                'x1': x1,
                'x2': x2,
                'high': high,
                'low': low,
                "trend": 0
            }
        elif dataframe['close'] < self.cx2om_info[metadata["pair"]]["low"]:
            self.cx2om_info[metadata["pair"]] = {
                'x1': x1,
                'x2': x2,
                'high': high,
                'low': low,
                "trend": -1
            }
        elif dataframe['close'] > self.cx2om_info[metadata["pair"]]["high"]:
            self.cx2om_info[metadata["pair"]] = {
                'x1': x1,
                'x2': x2,
                'high': high,
                'low': low,
                "trend": 1
            }
        return dataframe

    def populate_high_trend(self, dataframe: DataFrame,
                           metadata: dict) -> DataFrame:
        if self.cx2om_info[metadata["pair"]]["trend"] == 1:
            dataframe.loc[(), 'high'] = 1
        return dataframe

    def populate_low_trend(self, dataframe: DataFrame,
                            metadata: dict) -> DataFrame:
        if self.cx2om_info[metadata["pair"]]["trend"] == -1:
            dataframe.loc[(), 'low'] = 1
        return dataframe

**Comment by xmatthias** [comment]
You're reassigning all ohlcv candles - so all backtests done with this are invalid.
Please use new column names for comparisons - and best do a pandas tutorial, which will teach you how to work with pandas correctly.

**Comment by alitekdemir** [comment]
> You're reassigning all ohlcv candles - so all backtests done with this are invalid. Please use new column names for comparisons - and best do a pandas tutorial, which will teach you how to work with pandas correctly.

sorry i made a mistake.
I did find and replace while copying the code. So the dataframe["high"] looked different but I changed the code again. Please I am making a logic error. Give your opinion on this.

**Comment by xmatthias** [comment]
yes you do. you cannot use if/else - but need to use corresponding pandas methods.

As said - best look for a tutorial / search for it in the context of pandas - that way you'll learn how to do it (there's no benefit if i  do it for you - you'll still not have understood it next time).
